src/util.c

```diff
@@ -180,15 +180,13 @@ gboolean is_url(const gchar *string)
  * Glib-ish Utility functions.
  */
 
-gchar *trg_base64encode(const gchar *filename)
+gchar *trg_base64encode(const gchar *filename, GError **error)
 {
-    GError *error = NULL;
-    GMappedFile *mf = g_mapped_file_new(filename, FALSE, &error);
+    GMappedFile *mf = g_mapped_file_new(filename, FALSE, error);
     gchar *b64out = NULL;
 
-    if (error) {
-        g_error("%s", error->message);
-        g_error_free(error);
+    if (error && *error) {
+        return NULL;
```

Comment:
This can be in a bad state if `error == NULL && `mf == NULL`.

I would probably just put a `g_assert(error != NULL)` before it is used. This is an internal API and we should always be passing an error.

Reply:
This same comment applies to src/trg-file-parser.c

Reply:
I'm confused how that state could ever happen. I assume `error` can never be `NULL` but I added the check just because it seems weird to dereference it without an explicit check.